src/engine/ecs.cpp

```diff
@@ -1,10 +1,28 @@
 #include "ecs.hpp"
 #include "comp/game_entity.hpp"
+#include "comp/player_controlled.hpp"
+#include "system/physics.cpp"
+#include "system/player_controlled.cpp"
 
 ECS_IMPL_COMPONENT(CompGameEntity, game_entity)
+ECS_IMPL_COMPONENT(CompPlayerControlled, player_controlled)
 
-void ECS::update() {
+ECS::ECS() {
+  this->current_entity_id = 0;
+  this->systems.push_back(new SystemPhysics);
+  this->systems.push_back(new SystemPlayerControlled);
```

Comment:
Since we're looping from position 0 to the end of the list, the 'play controlled' system should come before the physics system (as it's adding the acceleration)